File: 251002/Samoilenko/publisher/src/main/java/com/example/discussion/repository/WriterRepository.java
package com.example.discussion.repository;

import com.example.discussion.model.Writer;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

//@Repository
public interface WriterRepository extends JpaRepository<Writer, Long> {
    Optional<Writer> findByLogin(String login);
}

File: 251003/Pankratiev/rest/src/main/kotlin/bsuir/dc/rest/repository/WriterRepository.kt
package bsuir.dc.rest.repository

import bsuir.dc.rest.entity.Writer
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface WriterRepository : JpaRepository<Writer, Long> {
    fun findByLoginIgnoreCase(login: String): Writer?
}